src/jsn.erl

```diff
@@ -656,6 +686,38 @@ from_proplist(X, _Options) -> X.
 %% internal functions
 %%==============================================================================
 
+-spec apply_condition(condition()|conditions(), json_term()) -> boolean().
+apply_condition({Path, ConditionFun}, Element) when is_function(ConditionFun, 1) ->
+    ConditionFun(get(Path, Element));
+apply_condition({Path, Value}, Element) ->
+    get(Path, Element) == Value;
+apply_condition(ConditionFun, Element) when is_function(ConditionFun, 1) ->
+    ConditionFun(Element);
+apply_condition([Condition|Rest], Element) ->
```

Comment:
I would also use `apply_conditions/2` here, to prevent incidentally supporting weird nesting of conditions. The second clause of `apply_conditions/2` can handle the case for passing through a singular condition to the `apply_condition/2` function, but I think it would be better from both a style and correctness perspective to avoid overloading `apply_condition/2` to support singular and plural conditions.